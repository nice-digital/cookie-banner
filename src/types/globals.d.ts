export {};

import { CookieControl } from "./cookie-control";

declare global {
	interface Window {
		dataLayer: Record<string, unknown>[];

		CookieControl: CookieControl;
	}
}
