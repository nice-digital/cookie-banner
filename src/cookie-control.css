#ccc-notify,
#ccc .ccc-notify-button {
	border-radius: 0 !important;
}

.ccc-notify-text {
	width: 100%; /* To fix IE11 layout bug */
}
