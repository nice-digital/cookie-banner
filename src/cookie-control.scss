@import '~@nice-digital/nds-core/scss/core.scss';

@mixin focus-ring {
  // Unfortunately we can't use @include nds-default-focus-style; because we
  // have to use important to override Cookie Control's styling
  outline: rem($nds-spacing-x-small) solid $nds-colour-focus !important;
}

#ccc {
  #ccc-module {
    animation-duration: 0.001ms !important;
  }

  #ccc-close {
    &:focus {
      @include focus-ring;
    }
  }

  #ccc-title {
    font-size: rem(nds-get-font-size(h4)) !important;
  }

  h3 {
    font-size: rem(nds-get-font-size(h5)) !important;
  }

  #ccc-content {
    a {
      color: $nds-colour-link !important;

      span {
        color: $nds-colour-link !important;
      }

      &:visited {
        color: $nds-colour-link-visited !important;
      }

      &:hover {
        color: $nds-colour-link-hover !important;
      }

      &:focus {
        @include focus-ring;
      }
    }
  }

  // Cookie Control uses icons with a fixed width, but we hide the external link one (see below)
  // So we want to also remove the width it leaves
  .ccc-svg-element {
    width: auto !important;

    // See https://designnotes.blog.gov.uk/2016/11/28/removing-the-external-link-icon-from-gov-uk/
    [data-icon='external-link'] {
      display: none;
    }
  }

  // Reject/accept buttons
  .ccc-notify-button {
    border-radius: 0 !important;
    padding: em($nds-spacing-small $nds-spacing-medium) !important;

    &:focus {
      @include focus-ring;
    }
  }

  // Accept button
  .ccc-accept-button {
    background: $nds-colour-btn-cta !important;
    border: em(1px) solid $nds-colour-btn-cta-text !important;

    span {
      color: $nds-colour-btn-cta-text !important;
    }

    &:hover,
    &:focus,
    &:active {
      background: $nds-colour-btn-cta-hover !important;
    }
  }

  // Reject button
  .ccc-reject-button {
    background: $nds-colour-btn-inverse !important;
    border: em(1px) solid $nds-colour-btn-inverse-text !important;

    &:hover,
    &:focus,
    &:active {
      background: $nds-colour-btn-inverse-hover !important;
    }
  }

  .checkbox-toggle {
    background: $nds-colour-nice-dark-grey !important;
    border-color: $nds-colour-nice-dark-grey !important;

    &:focus-within {
      @include focus-ring;
      outline-offset: rem($nds-spacing-xx-small);

      .checkbox-toggle-input:focus ~ .checkbox-toggle-toggle {
        outline: none !important;
      }
    }

    .checkbox-toggle-input:focus ~ .checkbox-toggle-toggle {
      @include focus-ring;
    }

    .checkbox-toggle-toggle {
      background: $nds-colour-nice-white !important;
    }

    .checkbox-toggle-on,
    .checkbox-toggle-off {
      color: $nds-colour-nice-white !important;
    }
  }
}
