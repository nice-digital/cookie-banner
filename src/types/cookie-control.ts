export interface CookieControl {
	load: (config: CookieControlConfig) => void;
	update: (config: CookieControlConfig) => void;
	open: () => void;
	hide: () => void;
	acceptAll: () => void;
	rejectAll: () => void;
	config: () => CookieControlConfig;
	geoInfo: () => GeoInfo;
	getCookie: () => string | false;
	getAllCookies: () => { [key: string]: string };
	saveCookie: (name: string, value: string, expiryLength: number) => void;
	delete: (name: string) => void;
	deleteAll: () => void;
	changeCategory: (index: number, grantConsent: boolean) => void;
	toggleCategory: (index: number) => void;
	getCategoryConsent: (index: number) => boolean | null;
	// Custom properties
	preferenceCookies: boolean;
	analyticsCookies: boolean;
}

interface GeoInfo {
	country: string;
	countryName: string;
	withinEU: boolean;
}

interface CookieControlText {
	title?: string;
	intro?: string;
	acceptSettings?: string;
	acceptRecommended?: string;
	accept?: string;
	rejectSettings?: string;
	reject?: string;
	settings?: string;
	necessaryTitle?: string;
	necessaryDescription?: string;
	thirdPartyTitle?: string;
	thirdPartyDescription?: string;
	on?: string;
	off?: string;
	notifyTitle?: string;
	notifyDescription?: string;
	closeLabel?: string;
	cornerButton?: string;
	landmark?: string;
}

export interface CookieControlPurpose {
	name: string;
	label: string;
	lawfulBasis?: "legitimate interest" | "consent";
	description: string;
	cookies: string[];
	onAccept?: () => void;
	onRevoke?: () => void;
	recommendedState?: boolean;
}

export interface CookieControlConfig {
	apiKey: string;
	product: string;
	necessaryCookies?: string[];
	optionalCookies?: CookieControlPurpose[];
	mode?: "gdpr" | "ccpa";
	consentCookieExpiry?: number;
	logConsent?: boolean;
	encodeCookie?: boolean;
	subDomains?: boolean;

	// Appearance and behaviour
	initialState?: "open" | "closed" | "notify" | "top" | "box";
	notifyOnce?: boolean;
	rejectButton?: false;
	layout?: "slideout" | "popup";
	position?: "left" | "right";
	theme?: "light" | "dark";
	toggleType?: "slider" | "checkbox";
	acceptBehaviour?: "all" | "recommended";
	closeOnGlobalChange?: boolean;
	closeStyle?: "icon" | "labelled" | "button";
	notifyDismissButton?: boolean;
	settingsStyle?: "button" | "link";
	setInnerHTML?: boolean;
	excludedCountries?: string[];
	locale?: string;
	onLoad?: () => void;
	sameSiteCookie?: boolean;
	sameSiteValue?: "Strict" | "Lax" | "None";

	// Accessibility
	accessibility?: {
		accessKey?: string;
		highlightFocus?: boolean;
		overlay?: boolean;
	};

	// Text
	text?: CookieControlText;

	// Branding
	branding?: {
		fontFamily?: string;
		fontSizeTitle?: string;
		fontSizeHeaders?: string;
		fontSize?: string;
		fontColor?: string;
		backgroundColor?: string;
		acceptText?: string;
		acceptBackground?: string;
		rejectText?: string;
		rejectBackground?: string;
		toggleText?: string;
		toggleColor?: string;
		toggleBackground?: string;
		alertText?: string;
		alertBackground?: string;
		buttonIcon?: string;
		buttonIconWidth?: number;
		buttonIconHeight?: number;
		removeIcon?: boolean;
		removeAbout?: boolean;
	};

	// Statement
	statement?: {
		description?: string;
		name?: string;
		url?: string;
		updated?: string;
	};

	// Locales
	locales?: {
		locale?: string;
		mode?: "gdpr" | "ccpa" | "hidden";
		location?: string[];
		optionalCookies: {
			name: string;
			label: string;
			description: string;
		}[];
		text?: CookieControlText;
	}[];
}

/**
 * The type of the cookie object that the Cookie Control plugin stores
 */
export interface CookieControlCookie {
	optionalCookies: {
		preferences?: "accepted" | "revoked";
		analytics?: "accepted" | "revoked";
	};
	// There are other properties like necessaryCookies, consentDate etc but we don't need them...
}
