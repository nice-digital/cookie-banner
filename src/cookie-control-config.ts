import { CookieControlConfig } from "./types/cookie-control";

export const cookieControlConfig: CookieControlConfig = {
	apiKey: "843970763414bd3ac1229c12767f5e7fab9ef68c",
	product: "PRO_MULTISITE",
	encodeCookie: true,
	onLoad: function (): void {
		// We need the try/catch because on first load, when the cookie doesn't exist then getCategoryConsent errors
		let preferenceCookies = false;
		let analyticsCookies = false;
		let marketingCookies = false;
		try {
			preferenceCookies = window.CookieControl.getCategoryConsent(0) || false;
			analyticsCookies = window.CookieControl.getCategoryConsent(1) || false;
			marketingCookies = window.CookieControl.getCategoryConsent(2) || false;
		} catch {
		} finally {
			window.CookieControl.analyticsCookies = analyticsCookies;
			window.CookieControl.preferenceCookies = preferenceCookies;
			window.CookieControl.marketingCookies = marketingCookies;
			window.dataLayer.push({
				event: "cookie.load",
				preferenceCookies,
				analyticsCookies,
				marketingCookies,
			});
		}
	},
	necessaryCookies: [
		// AWS sticky sessions - https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/elb-sticky-sessions.html
		"AWSELB",
		"AWSELBCORS",
		"AWSALBTG",
		"AWSALBTGCORS",
		// NICE Accounts
		"__ngat_1.2",
		"__nrpa_2.2",
		// IdAM
		"IdAM",
		"IdAM-*",
		// OpenAthens
		".AspNet.ClientCookie",
		"OpenIdConnect.nonce.*",
		// General ASP.NET
		"ASPXAUTH",
		"ASP.NET_SessionId",
		// .NET MVC CSRF
		"__RequestVerificationToken",
		// Evidence Search UI state
		"downloadopen",
		// NICE.Boostrap scroll position
		"ud",
		// INTERNAL USE
		// For InDev -> Consultations
		"documentReferrer",
		// Pathways authoring
		"display-node-orders",
		//EPPI
		"data-extraction-fixed-tables",
		"data-extraction-input-vertical",
		"browser-warning-shown",
		"compact",
		"dedupe-show-details",
		"relevant-highlight",
		"irrelevant-highlight",
		"dedupe-view",
		"text-highlighter-hide",
	],
	optionalCookies: [
		{
			name: "preferences",
			label: "Preference cookies",
			description:
				"These cookies remember information that changes the way our website behaves or looks, like your preferred layout or previously viewed pages.",
			cookies: [
				// Consultations: Preference for hiding the tutorial
				"TutorialVisible",
				// HDAS: Preference for search panel position
				"HoistSearches",
				// Guidance list: Preference for applied filters
				"NICE_guidanceList_*",
				// Pathways recently viewed and UI state
				"mobile-mode",
				"history-stack",
				"history-stack-groupUid",
				"history-stack-details",
			],
			onAccept: function (): void {
				window.CookieControl.preferenceCookies = true;
				window.dataLayer.push({
					event: "cookie.preferences.accept",
					preferenceCookies: true,
				});
			},
			onRevoke: function (): void {
				window.CookieControl.preferenceCookies = false;
				window.dataLayer.push({
					event: "cookie.preferences.revoke",
					preferenceCookies: false,
				});
			},
		},
		{
			name: "analytics",
			label: "Website usage cookies",
			description:
				"We use tools such as Google Analytics, Hotjar, VWO and Loop11 to help us anonymously measure how you use our websites. This allows us to make improvements based on our users' needs. These tools set cookies that store anonymised information about how you got to the site, and how you interact with the site.",
			cookies: [
				// Google Analytics
				"_ga",
				"_gat_UA-*",
				"_gid",
				// Hotjar
				"_hj*",
				"ajs_*",
				// VWO
				"_vwo*",
				"_vis*",
			],
			onAccept: function (): void {
				window.CookieControl.analyticsCookies = true;
				window.dataLayer.push({
					event: "cookie.analytics.accept",
					analyticsCookies: true,
				});
			},
			onRevoke: function (): void {
				window.CookieControl.analyticsCookies = false;
				window.dataLayer.push({
					event: "cookie.analytics.revoke",
					analyticsCookies: false,
				});

				// Cookie Control doesn't handle local storage automatically, so we have to tidy analytics local storage ourselves
				window.localStorage.removeItem("vwoSn");

				// HotJar uses local storage with keys starting _hj
				Object.keys(window.localStorage)
					.filter((key) => /^_hj/.test(key))
					.forEach((key) => window.localStorage.removeItem(key));
			},
		},
		{
			name: "marketing",
			label: "Marketing and advertising cookies",
			description:
				"We use Google Ads to serve adverts to users on Google. It uses cookies to help us measure how many times people click on these ads and interact with our site.",
			cookies: [
				// Conversion linker https://support.google.com/tagmanager/answer/7549390?hl=en
				"_gcl_*",
			],
			thirdPartyCookies: [
				{
					name: "Google Ads",
					optOutLink: "https://adssettings.google.com/",
				},
			],
			vendors: [
				{
					name: "Google Ads",
					description:
						"Google Ads serves adverts on Google products and services. It uses cookies help to make advertising more effective. Without cookies, itâ€™s harder for an advertiser to reach its audience, or to know how many ads were shown and how many clicks they received.",
					thirdPartyCookies: true,
					url: "https://policies.google.com/technologies/ads",
				},
			],
			onAccept: function (): void {
				window.CookieControl.marketingCookies = true;
				window.dataLayer.push({
					event: "cookie.marketing.accept",
					marketingCookies: true,
				});
			},
			onRevoke: function (): void {
				window.CookieControl.marketingCookies = false;
				window.dataLayer.push({
					event: "cookie.marketing.revoke",
					marketingCookies: false,
				});
			},
		},
	],

	// Styling
	initialState: "open",
	position: "left",
	theme: "light",
	rejectButton: true,
	toggleType: "slider",

	// Text
	text: {
		title: "Cookies on the NICE website and services",
		intro:
			"Cookies are files saved on your phone, tablet or computer when you visit a website.\nWe use cookies to store information about how you use the NICE website and services, such as the pages you visit.",
		notifyTitle: "Cookies on the NICE website and services",
		notifyDescription:
			"We store cookies on your device to make our site work and to help us understand how the site is being used.",
		settings: "Manage cookies",
		necessaryTitle: "Essential cookies",
		necessaryDescription:
			"These cookies enable basic functions like page navigation and access to secure areas of the website. Our website cannot function properly without these cookies and they can only be deactivated by changing your browser preferences.",
		acceptSettings: "Accept all cookies",
		rejectSettings: "Reject cookies",
		closeLabel: "Close cookie banner",
	},

	// Statement
	statement: {
		description: "For more information, view our",
		name: "cookie statement.",
		url: "https://www.nice.org.uk/cookies",
		updated: "16/10/2020",
	},

	// Branding
	branding: {
		fontFamily: "Lato,'Helvetica Neue',Helvetica,Arial,sans-serif",
		removeAbout: true,
		removeIcon: true,
		fontSize: "1rem",
		fontColor: "#0e0e0e",
		backgroundColor: "#fff",
	},

	// Accessibility
	accessibility: {
		// We set our own focus styles
		highlightFocus: false,
	},
};
