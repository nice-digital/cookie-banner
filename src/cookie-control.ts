import https from "https";

import { CookieControl as CookieControlType } from "./types/cookie-control";

interface ValLoaderReturn {
	cacheable: boolean;
	code: string;
}

const CookieControlScriptUrl =
	"https://cc.cdn.civiccomputing.com/9/cookieControl-9.3.1.min.js";

// Loads the civic cookie control banner from the Civic CDN
const getCivicCookieControl = async (): Promise<string> => {
	return new Promise(async (resolve, reject) => {
		const body: Uint8Array[] = [];
		const req = await https.get(CookieControlScriptUrl, (res) => {
			res.on("data", (chunk) => body.push(chunk));
			res.on("end", () => {
				const data = Buffer.concat(body).toString();
				resolve(data);
			});
		});
		req.on("error", (e) => {
			reject(e);
		});
		req.end();
	});
};

// Used by val-loader, at build time: val-loader looks for a default export
// We do this for 2 reasons:
//  a) improve load times by loading the Civic banner at build time
//	b) not to rely on an external CDN at run time
//  c) re-export the CookieControl object to allow 'proper' modules and not rely on a global reference defined on window
export default async function (): Promise<ValLoaderReturn> {
	const civicCookieControlScript = await getCivicCookieControl();

	return {
		cacheable: true,
		code: `${civicCookieControlScript}\nmodule.exports = { CookieControl: window.CookieControl }`,
	};
}

// Used purely for a type definition, hence the use of a placeholder, as the const needs to be initiated with something
// This is replaced by val-loader at build time - see the default exported function above
export const CookieControl: CookieControlType = ({
	load: () => {
		/* deliberately empty, used for mocking */
	},
} as unknown) as CookieControlType;
