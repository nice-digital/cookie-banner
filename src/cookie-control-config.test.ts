import { cookieControlConfig } from "./cookie-control-config";
import { CookieControl } from "./types/cookie-control";

describe("cookie-control-config", () => {
	beforeEach(() => {
		window.dataLayer = [];
		window.CookieControl = {
			getCategoryConsent: (index: number): boolean => index === 0,
		} as CookieControl;
	});

	it("should match snapshot", () => {
		expect(cookieControlConfig).toMatchSnapshot();
	});

	it("should push cookie.load dataLayer event on load", () => {
		cookieControlConfig.onLoad && cookieControlConfig.onLoad();
		expect(window.dataLayer).toHaveLength(1);
		expect(window.dataLayer[0]).toEqual({
			event: "cookie.load",
			preferenceCookies: true,
			analyticsCookies: false,
		});
	});

	it.each([
		["preferences.accept", "onAccept", "preferenceCookies", true, 0],
		["preferences.revoke", "onRevoke", "preferenceCookies", false, 0],
		["analytics.accept", "onAccept", "analyticsCookies", true, 1],
		["analytics.revoke", "onRevoke", "analyticsCookies", false, 1],
	])(
		"should pass %s event into the dataLayer %s %s",
		(
			eventName: string,
			methodName: string,
			cookieType: string,
			boolVal: boolean,
			index: number
		) => {
			const cookieTypeConfig = cookieControlConfig.optionalCookies?.[index];

			if (cookieTypeConfig) {
				const method = cookieTypeConfig[methodName as "onRevoke" | "onAccept"];
				if (method) method();
			}

			expect(window.dataLayer).toHaveLength(1);
			expect(window.dataLayer[0]).toEqual({
				event: `cookie.${eventName}`,
				[cookieType]: boolVal,
			});
		}
	);

	it.each([
		["preferenceCookies", true, "onAccept", 0],
		["preferenceCookies", false, "onRevoke", 0],
		["analyticsCookies", true, "onAccept", 1],
		["analyticsCookies", false, "onRevoke", 1],
	])(
		"should set CookieControl.%s variable to %s %s",
		(
			variableName: string,
			boolVal: boolean,
			methodName: string,
			index: number
		) => {
			const cookieTypeConfig = cookieControlConfig.optionalCookies?.[index];

			if (cookieTypeConfig) {
				const method = cookieTypeConfig[methodName as "onRevoke" | "onAccept"];
				if (method) method();
			}

			expect(
				window.CookieControl[
					variableName as "preferenceCookies" | "analyticsCookies"
				]
			).toEqual(boolVal);
		}
	);
});
