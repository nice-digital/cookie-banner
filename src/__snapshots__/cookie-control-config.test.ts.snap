// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cookie-control-config should match snapshot 1`] = `
Object {
  "accessibility": Object {
    "highlightFocus": false,
  },
  "apiKey": "843970763414bd3ac1229c12767f5e7fab9ef68c",
  "branding": Object {
    "backgroundColor": "#fff",
    "fontColor": "#0e0e0e",
    "fontFamily": "Lato,'Helvetica Neue',Helvetica,Arial,sans-serif",
    "fontSize": "1rem",
    "removeAbout": true,
    "removeIcon": true,
  },
  "encodeCookie": true,
  "initialState": "open",
  "necessaryCookies": Array [
    "AWSELB",
    "AWSELBCORS",
    "AWSALBTG",
    "AWSALBTGCORS",
    "__ngat_1.2",
    "__nrpa_2.2",
    "IdAM",
    "IdAM-*",
    ".AspNet.ClientCookie",
    "OpenIdConnect.nonce.*",
    "ASPXAUTH",
    "ASP.NET_SessionId",
    "__RequestVerificationToken",
    "downloadopen",
    "ud",
    "documentReferrer",
    "display-node-orders",
    "data-extraction-fixed-tables",
    "data-extraction-input-vertical",
    "browser-warning-shown",
    "compact",
    "dedupe-show-details",
    "relevant-highlight",
    "irrelevant-highlight",
    "dedupe-view",
    "text-highlighter-hide",
    "evercookie_cache",
    "evercookie_etag",
    "evercookie_png",
    ".ukps",
    ".ukpsLease",
  ],
  "onLoad": [Function],
  "optionalCookies": Array [
    Object {
      "cookies": Array [
        "TutorialVisible",
        "HoistSearches",
        "NICE_guidanceList_*",
        "mobile-mode",
        "history-stack",
        "history-stack-groupUid",
        "history-stack-details",
      ],
      "description": "These cookies remember information that changes the way our website behaves or looks, like your preferred layout or previously viewed pages.",
      "label": "Preference cookies",
      "name": "preferences",
      "onAccept": [Function],
      "onRevoke": [Function],
    },
    Object {
      "cookies": Array [
        "_ga",
        "_gat_UA-*",
        "_gid",
        "_hj*",
        "ajs_*",
        "_vwo*",
        "_vis*",
      ],
      "description": "We use tools such as Google Analytics, Hotjar, VWO and Loop11 to help us anonymously measure how you use our websites. This allows us to make improvements based on our users' needs.

These tools set cookies that store anonymised information about how you got to the site, and how you interact with the site.",
      "label": "Website usage cookies",
      "name": "analytics",
      "onAccept": [Function],
      "onRevoke": [Function],
    },
    Object {
      "cookies": Array [
        "_gcl_*",
        "li_fat_id",
      ],
      "description": "We use Google Ads and LinkedIn to show adverts on external sites to promote NICE services, events and content.

These services use cookies to help make advertising more effective. These cookies are used for things like showing relevant adverts based on website visits, preventing the same ad from continuously re-appearing, or by measuring how many times people click on these adverts.",
      "label": "Marketing and advertising cookies",
      "name": "marketing",
      "onAccept": [Function],
      "onRevoke": [Function],
      "thirdPartyCookies": Array [
        Object {
          "name": "Google Ads",
          "optOutLink": "https://adssettings.google.com/",
        },
        Object {
          "name": "LinkedIn",
          "optOutLink": "https://www.linkedin.com/help/linkedin/answer/62931/manage-advertising-preferences?lang=en",
        },
      ],
      "vendors": Array [
        Object {
          "description": "Google Ads serves adverts on Google products and services. It uses cookies to help to make advertising more effective.",
          "name": "Google Ads",
          "thirdPartyCookies": true,
          "url": "https://policies.google.com/technologies/ads",
        },
        Object {
          "description": "LinkedIn shows adverts when you browse LinkedIn products and services. It uses cookies to help to make advertising more effective.",
          "name": "LinkedIn",
          "thirdPartyCookies": true,
          "url": "https://www.linkedin.com/legal/cookie-policy",
        },
      ],
    },
  ],
  "position": "left",
  "product": "PRO_MULTISITE",
  "rejectButton": true,
  "statement": Object {
    "description": "For more information, view our",
    "name": "cookie statement.",
    "updated": "22/10/2020",
    "url": "https://www.nice.org.uk/cookies",
  },
  "text": Object {
    "acceptSettings": "Accept all cookies",
    "closeLabel": "Close cookie banner",
    "intro": "Cookies are files saved on your phone, tablet or computer when you visit a website.
We use cookies to store information about how you use the NICE website and services, such as the pages you visit.",
    "necessaryDescription": "These cookies enable basic functions like page navigation and access to secure areas of the website. Our website cannot function properly without these cookies and they can only be deactivated by changing your browser preferences.",
    "necessaryTitle": "Essential cookies",
    "notifyDescription": "We store cookies on your device to make our site work and to help us understand how the site is being used.",
    "notifyTitle": "Cookies on the NICE website and services",
    "rejectSettings": "Reject cookies",
    "settings": "Manage cookies",
    "title": "Cookies on the NICE website and services",
  },
  "theme": "light",
  "toggleType": "slider",
}
`;
